src/Arcus.WebApi.OpenApi.Extensions/OAuthAuthorizeOperationFilter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Arcus.WebApi.OpenApi.Extensions
+{
+    /// <summary>
+    /// A Swashbuckle operation filter that adds OAuth security definitions to authorized API operations.
+    /// </summary>
+    public class OAuthAuthorizeOperationFilter : IOperationFilter
+    {
+        private readonly IEnumerable<string> _scopes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OAuthAuthorizeOperationFilter"/> class.
+        /// </summary>
+        public OAuthAuthorizeOperationFilter(IEnumerable<string> scopes)
+        {
+            _scopes = scopes;
+        }
+
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            var hasAuthorize = context.MethodInfo.GetCustomAttributes(true).OfType<AuthorizeAttribute>().Any() ||
```

Comment:
Also, the `DeclaringType` could be `null` and should be checked.

I also think that using the `.GetCustomAttribute<>` would be more useful here.
This returns `null` when no such attribute can be found.

And maybe (not blocking) the `hasAuthorize` should be  split in two booleans: `operationHasAuthorize` and `typeHasAuthorize` or some sorts.

Reply:
In theory, it can indeed be null and considering the defensive programming techniques it should indeed be checked.
However, I wonder if you can have a method in .NET that hasn't been declared in a type :)

Reply:
The `GetCustomAttribute` extension method calls the `GetCustomAttributes` method behind the scenes.  I wouldn't consider it as a better way to accomplish what we're doing here.

Reply:
Well, we won't have to call `.Any` and `.OfType` three times anymore. That was what I was thinking.
Not blocking of course.

Reply:
Code has been changed, thanks @fgheysels 	!